rest/server_context.go

```diff
@@ -372,6 +372,17 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(config DatabaseConfig, useE
 		return nil, err
 	}
 
+	secure := spec.IsTLS()
+	if sc.config.Bootstrap.AllowInsecureServerConnections == nil || !*sc.config.Bootstrap.AllowInsecureServerConnections {
+		if !secure && !spec.IsWalrusBucket() {
+			return nil, fmt.Errorf("couchbase server URL must use secure protocol when disallowing insecure server connections. Current URL: %s", spec.Server)
```

Comment:
"disallowing insecure connections" is the default here, so again the double negative is just going to cause confusion for users encountering this (they won't know about that option).


```suggestion
			return nil, fmt.Errorf("Must use secure scheme in Couchbase Server URL, or opt out using allow_insecure_server_connections: %s", spec.Server)
```